tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/core/SentenceTagger.scala

```diff
@@ -251,22 +253,37 @@ case object TokenPositionTagger extends SentenceTagger {
 case class NLPStackPostagger(baseTagger: Postagger, useCoarseTags: Boolean) extends SentenceTagger {
 
   override def tag(
-    sentence: Sentence
+    sentence: Sentence, constraints: Set[TransitionConstraint]
   ): TaggedSentence = {
 
+    val tokenToRequestedCpos: Map[Int, Symbol] = {
+      constraints flatMap {
+        case RequestedCpos(tokenIndex, cpos) =>
+          Some((tokenIndex, cpos))
+        case _ =>
+          None
+      }
+    }.toMap
     val taggedTokens: Map[Int, PostaggedToken] =
       Util.getPostaggedTokens(sentence, baseTagger)
-    val tagMap: Map[Int, Set[TokenTag]] = taggedTokens mapValues { tagged =>
-      Set(
-        if (useCoarseTags) {
-          TokenTag(
-            'autoCpos,
-            Symbol(WordClusters.ptbToUniversalPosTag.getOrElse(tagged.postag, "X"))
-          )
-        } else {
-          TokenTag('autoPos, Symbol(tagged.postag))
-        }
-      )
+    val tagMap: Map[Int, Set[TokenTag]] = taggedTokens map {
+      case (tokenIndex, tagged) =>
+        (tokenIndex,
+          if (useCoarseTags) {
+            Set(
+              TokenTag(
+                'autoCpos,
+                tokenToRequestedCpos.getOrElse(
+                  tokenIndex,
+                  Symbol(WordClusters.ptbToUniversalPosTag.getOrElse(tagged.postag, "X"))
+                )
+              )
+            )
+          } else if (tokenToRequestedCpos.contains(tokenIndex)) {
+            Set[TokenTag]()
```

Comment:
Shouldn't this use the value in `tokenToRequestedCpos` corresponding to the `tokenIndex`? It seems weird to be empty, but I might be misunderstanding soemthing.